monkey/infection_monkey/monkey.py

```diff
@@ -291,6 +291,11 @@ def cleanup(self):
 
         logger.info("Monkey is shutting down")
 
+    def _propagation_enabled(self) -> bool:
+        # If self._current_depth is None, assume that propagation is desired.
+        # Actual max depth will be retrieved from the server later.
```

Comment:
So the tunnel is started, but when it's retrieved from the server and it turns out to be >1 the tunnel thread is never stopped/joined?

Reply:
`self._current_depth` is not updated, so this condition doesn't occur. I've updated the comment to help clarify.